main.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+module Enumerable
+  # 1. my_each method
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    size.times { |index| yield array[index] }
+    self
+  end
+
+  # 2. my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    pos = 0
+    while pos < arr.length
+      yield(arr[pos], pos)
+      pos += 1
+    end
+
+    self
+  end
+
+  # 3.my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    to_a.my_each { |item| new_arr << item if yield(item) }
+    new_arr
+  end
+
+  # 4.my_all?
+
+  def my_all?(args = nil)
+    if block_given?
+      counter_false = 0
+      my_each { |num| counter_false += 1 unless yield num }
+      counter_false.zero?
+    elsif args.nil
+      my_all? { |num| num }
+    else
+      my_all? { |num| args == num }
+    end
+  end
+
+  # 5.my_any?
+
+  def my_any?(args = nil)
+    if block_given?
+      counter_true = 0
+      my_each { |num| counter_true += 1 if yield num }
+      counter_true.positive?
+    elsif args.nil?
+      my_any? { |num| num }
+    else
+      my_any? { |num| args == num }
+    end
+  end
+end
```

Comment:
my_any?

when no block or argument is given returns true if at least one of the collection is not false or nil
`p [nil, false, nil, false].my_any?` - should return false

when a class is passed as an argument returns true if at least one of the collection is a member of such class
`p [12,3,34].my_any?(Integer)` should return true

when a Regex is passed as an argument returns false if none of the collection matches the Regex
`p ['ghetto','wild'].my_any?(/z/)` should return false

when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`p ['cat',2,32,].my_any?('cat')` should return true